src/background/utils/preinject.js

```diff
@@ -279,18 +279,19 @@ function prepareScript(script) {
   // adding `;` on a new line in case some required script ends with a line comment
   const reqsSlices = reqs ? [].concat(...reqs.map(req => [req, '\n;'])) : [];
   const hasReqs = reqsSlices.length;
+  const wrap = !meta.unwrap;
   const injectedCode = [
     // hiding module interface from @require'd scripts so they don't mistakenly use it
-    `window.${dataKey}=function(${dataKey}){try{with(this)((define,module,exports)=>{`,
+    wrap ? `window.${dataKey}=function(${dataKey}){try{with(this)((define,module,exports)=>{` : '',
     ...reqsSlices,
     // adding a nested IIFE to support 'use strict' in the code when there are @requires
-    hasReqs ? '(()=>{' : '',
+    hasReqs && wrap ? '(()=>{' : '',
     code,
     // adding a new line in case the code ends with a line comment
-    code.endsWith('\n') ? '' : '\n',
-    hasReqs ? '})()' : '',
+    code.endsWith('\n') || !wrap ? '' : '\n',
+    hasReqs && wrap ? '})()' : '',
     // 0 at the end to suppress errors about non-cloneable result of executeScript in FF
-    `})()}catch(e){${dataKey}(e)}};0`,
+    wrap ? `})()}catch(e){${dataKey}(e)}};0` : '',
```

Comment:
I guess `\n;0` at the end is still needed for Firefox (not tested though).

Reply:
Fixed.